### PR TITLE
LPS-187428 BrowserSniffer UpgradeProcess

### PR DESCRIPTION
**Pull request by alberttgomes**
[LPS-187428](https://liferay.atlassian.net/browse/LPS-187428)

### CODE REVIEW

**Comment by alberttgomes** [comment]
ci:test:sf

**Comment by alberttgomes** [comment]
ci:test:relevant

**Comment by alberttgomes** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+			_multiVMPool.clear();
+		}
+	}
+
+	@Inject
+	protected DDMTemplate ddmTemplate;
+
+	@Inject
+	protected FragmentEntry fragmentEntry;
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java:128]: @alberttgomes, no need to use them as `protected`, please keep them as `private`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	private Group _group;
+
+	@Inject
+	private MultiVMPool _multiVMPool;
+
+	@Inject
+	private Portal _portal;
+
+	@Inject
+	private ServiceContext _serviceContext;
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java:154]: @alberttgomes, I don't think that we need to annotate this with `@Inject`, actually we shouldn't use any annotation. Can you please give it a try?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_3_2/test/TemplateBrowserSnifferUpgradeProcessTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+		Assert.assertEquals(
+			read(".v5_3_2/expected-ddm-template-browser-sniffer-content.ftl"),
+			ddmTemplate.getScript());
+
+		fragmentEntry = getFragmentEntry();
+
+		Assert.assertEquals(
+			read(
+				".v5_3_2/expected-fragment-entry-browser-sniffer-content.html"),
+			fragmentEntry.getHtml());
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_3_2/test/TemplateBrowserSnifferUpgradeProcessTest.java:64]: @alberttgomes, try the following code instead:
```
Assert.assertEquals(
	read(".v5_3_2/expected-ddm-template-browser-sniffer-content.ftl"),
	getDDMTemplate().getScript());

Assert.assertEquals(
	read(".v5_3_2/expected-fragment-entry-browser-sniffer-content.html"),
	getFragmentEntry().getHtml());
```
If SF complain about it then declare local variables:
```
DDMTemplate ddmTemplate = getDDMTemplate();

Assert.assertEquals(
	read(".v5_3_2/expected-ddm-template-browser-sniffer-content.ftl"),
	ddmTemplate.getScript());

FragmentEntry fragmentEntry = getFragmentEntry();

Assert.assertEquals(
	read(".v5_3_2/expected-fragment-entry-browser-sniffer-content.html"),
	fragmentEntry.getHtml());
```

**Review by natocesarrego — CHANGES_REQUESTED** [review]
We're close, Albert. :slightly_smiling_face: 

I believe that the next pull will be the one to be forwarded to the external review. :smile:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java (hunk 1 above)
Comment (alberttgomes) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java:128]: No? we is using the ddmTemplate and fragmentEntry in `TemplateBrowserSnifferUpgradeProcessTest` ? like suggest here [link](https://github.com/natocesarrego/liferay-portal/pull/1313/commits/7864853a30b42e904540b4faa3f566c092156102#r1256358878)

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java (hunk 2 above)
Comment (alberttgomes) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java:154]: Sure!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/BaseTemplateUpgradeProcess.java [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+						updatePreparedStatement.addBatch();
+					}
+
+					updatePreparedStatement.executeBatch();
+				}
+			}
+		}
+	}
+
+	private static final Pattern _isAssignEmptyDDMTEmplatePattern =
```
[the remark this answers is not in the document]
Comment (alberttgomes) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/BaseTemplateUpgradeProcess.java:126]: So, how to we idenfier differences between `FragmentEntry` and `DDMTemplate` to `isAssignEmpty`?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_3_2/test/TemplateBrowserSnifferUpgradeProcessTest.java (hunk 3 above)
Comment (alberttgomes) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_3_2/test/TemplateBrowserSnifferUpgradeProcessTest.java:64]: hey @natocesarrego, the SF which is formated this way.
So based on this second exemple, this response the first comment above, right?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/BaseTemplateUpgradeProcess.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/BaseTemplateUpgradeProcess.java:126]: @alberttgomes, my point was regarding the capitalized `E` only, is it clearer?

You named the variable as `_isAssignEmptyDDMTEmplatePattern`, but it should be `_isAssignEmptyDDMTemplatePattern`

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java (hunk 1 above)
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/service/test/BaseTemplateUpgradeProcessTestCase.java:128]: I see your point, @alberttgomes. My code snippet wasn't 100% precise, sorry about that. As a good practice of OOP (Objects Oriented Programming), we should encapsulate the class attributes and access them through its methods get and set, am I clearer now?

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_3_2/test/TemplateBrowserSnifferUpgradeProcessTest.java (hunk 3 above)
Comment (natocesarrego) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/upgrade/v5_3_2/test/TemplateBrowserSnifferUpgradeProcessTest.java:64]: If SF is not accepting the first suggestion, please declare local variables as suggested above. :wink: 

Thanks

**Pull request closed without merging**